DotNetHowToSecurityTrimming/DotNetHowToSecurityTrimming/MicrosoftGraphHelper.cs

```diff
@@ -0,0 +1,205 @@
+﻿#define HowToExample
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+
+namespace DotNetHowToSecurityTrimming
+{
+    public class MicrosoftGraphHelper
+    {
+        private GraphServiceClient _graph;
+        private string _clientId;
+        private string _token;
+
+        public MicrosoftGraphHelper(string clientId)
+        {
+            _clientId = clientId;
+        }
+
+        public void CreateGraphServiceClient()
+        {
+            PublicClientApplication app = new PublicClientApplication(_clientId);
+            string[] scopes = { "User.ReadWrite.All", "Group.ReadWrite.All", "Directory.ReadWrite.All" };
+
+            // Instantiate the Microsoft Graph, and provide a way to acquire the token. If token expires, it will
+            // be acquired again
+            _graph = new GraphServiceClient(new DelegateAuthenticationProvider(
+            async (requestMessage) =>
+            {
+                AuthenticationResult result = null;
+                // If a user has already signed-in, we try first to acquire the token silently, and then if this fails
+                // we try to acquire it with a user interaction.
+                var user = app.Users.FirstOrDefault();
```

Comment:
yes, users is a list of user, since in this app only one user is signed in, we know that there is only one user at any given time and therefore choosing this user.

Reply:
I see, should it be SingleOrDefault()? I'd be to add your explanation to the comment

Reply:
@revitalbarletz , why not SingleOrDefault?